python/ray/air/config.py

```diff
@@ -149,31 +148,46 @@ class DatasetConfig:
     # True by default for the "train" dataset only.
     required: Optional[bool] = None
 
-    # Whether the dataset can be streamed into memory using pipelined reads.
-    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
-    # Note that streaming isn't enabled unless you set stream_window_size too.
-    # False by default for all datasets.
-    streamable: Optional[bool] = None
-
     # Whether to transform the dataset with the fitted preprocessor. This must be
     # enabled at least for the dataset that is fit.
     # True by default for all datasets.
     transform: Optional[bool] = None
 
-    # List of fields that the user cannot override in ``dataset_config``.
-    _noncustomizable_fields: Tuple[str] = ()
+    # Whether the dataset should be streamed into memory using pipelined reads.
+    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
+    # The amount of memory to use is controlled by `stream_window_size`.
+    # False by default for all datasets.
+    use_stream_api: Optional[bool] = None
```

Comment:
Nit: I feel like `use_stream_api` is more verbose than it needs to be, any opposition to something like `streaming` or `use_streaming`?
```suggestion
    streaming: Optional[bool] = None
```

Reply:
Hmm, my rationale here was to call out that this is changing the class returned by "get_dataset_shard()", which is changing more than just the underlying impl.

Not sure if users get this though.

Reply:
Ah, got it, thanks for adding that to the comment above!